apex/optimizers/fused_mixed_precision_lamb.py

```diff
@@ -0,0 +1,251 @@
+import torch
+from copy import deepcopy
+from itertools import chain
+from collections import defaultdict, abc as container_abcs
+
+from apex.multi_tensor_apply import multi_tensor_applier
+
+class FusedMixedPrecisionLamb(torch.optim.Optimizer):
+
+    def __init__(self, params, lr=1e-3, step=0, bias_correction=True,
+                 betas=(0.9, 0.999), eps=1e-6, weight_decay=0.01,
+                 amsgrad=False, adam_w_mode=True,
+                 grad_averaging=True, max_grad_norm=1.0, use_nvlamb=False,
+                 reduced_precision_dtype=None):
+        if amsgrad:
+            raise RuntimeError('FusedLAMB does not support the AMSGrad variant.')
+
+        # The learning rate (lr) and optimizer step (step) should be located on device
+        # in order to faciliated device sync free execution
+        defaults = dict(lr=torch.tensor(lr, dtype=torch.float32, device=torch.cuda.current_device()),
+                        step=torch.tensor([step], dtype=torch.int, device=torch.cuda.current_device()),
```

Comment:
Using `torch.cuda.current_device()` can lead to the failure of [`test_multi_device`](https://github.com/NVIDIA/apex/pull/1237/files#diff-f16fafddd5bfd009eaf6d3fec48df46f50326c2d9a25c829d6e544fee49447e1R289).

A fix might be to send these two tensors to the device [here](https://github.com/NVIDIA/apex/pull/1237/files#diff-fe6f34523dea6871f255b052775b32df8f6af5bae5dae0d2497f190b60c24beaR156) lazily in `step` method.

Reply:
I used the device of the parameters that are passed in and that seems to work.

Reply:
The mixed device test now passes.